### ISSUE TITLE
Colorspace tetrahedral interpolation incorrect in lut3d_interp_tetrahedral()

### ISSUE THREAD

**Issue by AdeMiller** [rigaya/NVEnc#512]
I'm pretty sure the following is incorrect:
```
        } else if (scalez > scalex) {
            scale0 = scaley;
            scale1 = scalez;
            scale2 = scalex;
            xA = x0; yA = y1; zA = z0;
            xB = x0; yB = y1; zB = z1;
        } else {
            scale0 = scaley;
            scale1 = scalez;
            scale2 = scalex;
            xA = x0; yA = y1; zA = z0;
            xB = x1; yB = y1; zB = z0;
        }
```
Note how `scale0` `scale1` and `scale2` are identical for both cases. In this case I believe the values of `scale1` and `scale2` should be flipped.

See: https://github.com/rigaya/NVEnc/pull/513

I've not tested this against your codebase. I discovered this issue while using your code as an example when implementing my own colorspace converter for which I have a lot of tests against both trilinear and tetrahedral interpolation. 

Thanks for providing such a useful set of examples for colorspace manipulation.

**Comment by rigaya** [comment]
Thank you for the pull request, I have just merged it. 

> In this case I believe the values of scale1 and scale2 should be flipped.

Yes, you are right, I must have copy pasted and then forgot to change to the correct variable...

> Thanks for providing such a useful set of examples for colorspace manipulation.

Thank you to you too, to letting me know and creating the pull request for the issue!

**Comment by rigaya** [comment]
Fix has been applied in [NVEnc 7.31](https://github.com/rigaya/NVEnc/releases/tag/7.31), so I'll close the issue.

Thanks again for the pull request.
